File: SINGLY_LINKED_LIST.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None


class SinglyLinkedList:
    def __init__(self):
        self.head = None

    def insert_first(self, x):
        new_node = Node(x)
        new_node.next = self.head
        self.head = new_node

    def insert_last(self, x):
        new_node = Node(x)

        if self.head is None:
            self.head = new_node
            return

        curr = self.head

        while curr.next:
            curr = curr.next

        curr.next = new_node

    def insert_middle(self, pos, x):
        new_node = Node(x)

        curr = self.head
        prev = self.head

        while curr.next:
            if curr.data == pos:
                break
            prev = curr
            curr = curr.next

        new_node.next = prev.next
        prev.next = new_node

    def delete_first(self):
        curr = self.head
        self.head = curr.next

    def delete_last(self):
        curr = self.head
        prev = self.head

        while curr.next:
            prev = curr
            curr = curr.next

        prev.next = None

    def delete_middle(self, pos):
        curr = self.head
        prev = self.head

        while curr.next:
            if curr.data == pos:
                break
            prev = curr
            curr = curr.next

        prev.next = curr.next

        print(prev.data)
        print(curr.data)

    def display(self):
        curr = self.head
        ans = []

        while curr:
            ans.append(curr.data)
            curr = curr.next

        print(ans)


# Driver Code
n = SinglyLinkedList()

n.insert_first(10)
n.insert_first(20)
n.insert_last(2)
n.insert_last(13)
n.insert_last(15)
n.insert_last(56)
n.insert_first(77)

n.insert_middle(13, 78)

# n.delete_first()
# n.delete_last()
# n.delete_middle(56)
# n.delete_middle(15)

n.display()

File: SinglyLinkedList.java
import java.util.*;
public class SinglyLinkedList {
    Node head;

    static class Node {
        int data;
        Node next;

        Node(int d) {
            data = d;
            next = null;
        }
    }

    //INSERT FIRST NODE
    public void insert(int data) {
        Node newNode = new Node(data);
        if (head == null) {
            head = newNode;
            return;
        }
        Node current = head;
        while (current.next != null) {
            current = current.next;
        }
        current.next = newNode;
    }

    public void printList() {
        Node current = head;
        while (current != null) {
            System.out.print(current.data + " ");
            current = current.next;
        }
        System.out.println();
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        SinglyLinkedList list = new SinglyLinkedList();
        list.insert(sc.nextInt());
        list.insert(sc.nextInt());
        list.insert(sc.nextInt());
        list.insert(sc.nextInt());

        System.out.println("Singly Linked List:");
        list.printList();
    }
}